src/icpp/utils.py

```
@@ -1,11 +1,50 @@
     func(path)
+
+
+def rmtree_force(path: Union[str, Path]) -> None:
+    """Like shutil.rmtree, but clears readonly bits on the way down and
+    silences the Python 3.12 deprecation of `onerror`.
+
+    `shutil.rmtree`'s `onerror` parameter was deprecated in 3.12 in favor
+    of `onexc` (with a different 3rd-arg signature). pylint 3.x running
+    against Python 3.13's stdlib stubs flags `onerror=` calls as
+    `W4903 deprecated-argument`. To keep CI clean across the
+    Python 3.10–3.14 matrix we branch by version here. `remove_readonly`
```

Ruff (0.15.12)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)